src/connection.h

```diff
@@ -58,6 +64,14 @@ typedef enum {
 #define CONN_FLAG_WRITE_BARRIER (1 << 1)        /* Write barrier requested */
 #define CONN_FLAG_ALLOW_ACCEPT_OFFLOAD (1 << 2) /* Connection accept can be offloaded to IO threads. */
 
+typedef enum {
+    CONN_TYPE_ID_NONE = 0,
```

Comment:
nit. Invalid seems clearer.

```suggestion
    CONN_TYPE_ID_INVALID = 0,
```